lsp-completion.el

```diff
@@ -501,7 +502,8 @@ The MARKERS and PREFIX value will be attached to each candidate."
           ;; always be shown
           ((equal action 'lambda) nil)
           ;; retrieve candidates
-          ((equal action t) (funcall candidates))))
+          ((equal action t)
```

Comment:
Why not use complete-with-action? This will be shorter and equivalent to what you have now. I think you had that in the version of this PR after my first review.

Reply:
Furthermore you did not update the null action (try-completion). I would simply go with complete-with-action and be done. Is there a point in the current implementation?

Reply:
~~~ elisp
(lambda (str pred action) ;; compliant completion table
  (if (eq action 'metadata)
      '(metadata (category . lsp-capf)
                 (cycle-sort-function . identity)
                 (display-sort-function . identity))
    (complete-with-action action (funcall candidates) str pred)))
~~~

Reply:
There are, you can see that in the comment for each case there.

Reply:
I would rather not risk changing all of them to use the `complete-with-action`, if you think the implementation of some certain part is not working with your case, please state that

Reply:
Okay fair enough. (My reason would be: I don't trust the implementation. So I have to review it point by point. In comparison complete-with-action is already provided and does the right thing. Furthermore why not go with the simpler code, why make things complicated?)